### PR TITLE
#587 Move configs from front to backend

### LINKED ISSUE #587: Remove config.js from repository cause of insecure.
Instead, call the server for these settings and store them in redux.
On the server-side can store then in key-vault

### PR DESCRIPTION
**Pull request by stannydan**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)

## Code reviewers

- [ ] @iuriikhrystiuk

### Second Level Review

- [ ] @sand0

## Summary of issue

We have some sensitive data in frontend

## Summary of change

Configs move to backend

## Testing approach

Not tested yet

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EventsExpress/ClientApp/src/components/event/share/ShareButtons.js [hunk 1]
```diff
@@ -23,3 +24,14 @@ export default class ShareButtons extends Component {
         );
     }
 }
+
+const mapStateToProps = (state) => {
+    return {
+        config: state.config
+    }
+};
+const mapDispatchToProps = dispatch => ({
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/event/share/ShareButtons.js]: This mapdispatchtoprops is not needed here.

File: EventsExpress/Controllers/FrontConfigsController.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        private readonly IOptions<FrontConfigsViewModel> _frontConfigs;
+
+        public FrontConfigsController(IOptions<FrontConfigsViewModel> frontConfigs)
+        {
+            _frontConfigs = frontConfigs;
+        }
+
+        public IActionResult GetConfigs()
+        {
+            _ = new FrontConfigsViewModel();
```
Comment (iuriikhrystiuk) [inline EventsExpress/Controllers/FrontConfigsController.cs]: Why do you need this discard value? Remove please.

File: EventsExpress/appsettings.json [hunk 3]
```diff
@@ -26,6 +26,14 @@
     "GoogleClientId": "",
     "GoogleClientSecret": ""
   },
+  "FrontEndConfigs": {
+    "FacebookClientId": "2741444662848625",
+    "GoogleClientId": "3429495463-t3uma25030ea0q819p2f600bi9j1869l.apps.googleusercontent.com",
+    "TwitterCallbackUrl": "/Authentication/TwitterLogin",
+    "TwitterConsumerKey": "9SrdZyy2wdtjVyXBpVK5blfCf",
+    "TwitterConsumerSecret": "dsTaBtiBcXSCK3AcyfM9OjO1QUHDlTBnKxqsmIOhqDPKRv6fax",
```
Comment (iuriikhrystiuk) [inline EventsExpress/appsettings.json]: Please, remove secrets from appsettings.

File: EventsExpress/ClientApp/src/components/event/share/ShareButtons.js (hunk 1 above)
Comment (stannydan) [reply EventsExpress/ClientApp/src/components/event/share/ShareButtons.js]: Remove

File: EventsExpress/Controllers/FrontConfigsController.cs (hunk 2 above)
Comment (stannydan) [reply EventsExpress/Controllers/FrontConfigsController.cs]: Deleted!

File: EventsExpress/appsettings.json (hunk 3 above)
Comment (stannydan) [reply EventsExpress/appsettings.json]: Done

File: EventsExpress/.config/dotnet-tools.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "version": 1,
```
Comment (iuriikhrystiuk) [inline EventsExpress/.config/dotnet-tools.json]: Undo this file or add to gitignore.

**Review by iuriikhrystiuk — DISMISSED** [review]

**Review by iuriikhrystiuk — APPROVED** [review]

**Pull request merged**